src/libcore/intrinsics.rs

```diff
@@ -547,6 +547,15 @@ extern "rust-intrinsic" {
     pub fn u64_mul_with_overflow(x: u64, y: u64) -> (u64, bool);
 }
 
+#[cfg(not(stage0))]
+extern "rust-intrinsic" {
+    /// Returns (a + b) mod 2^N, where N is the width of N in bits.
+    pub fn overflowing_add<T>(a: T, b: T) -> T;
```

Comment:
Any reason these are overflowing_* while the methods are wrapping_*? I think wrapping_* is a bit more descriptive, fwiw.